File: core/src/main/java/com/aps/api/model/comum/ParametrosConsulta.java
package com.aps.api.model.comum;

import java.io.Serializable;
import java.util.ArrayList;

import lombok.Data;

@Data
public class ParametrosConsulta implements Serializable {

	private static final long serialVersionUID = -6043995776305644426L;
	
	// filtros
	private Long id;
	private String nome;
	private ArrayList<Long> idLegislatura;
	private ArrayList<String> siglaUf;
	private ArrayList<String> siglaPartido;
	private String siglaSexo;
	private ArrayList<Long> ano;
	private ArrayList<Long> mes;
	private String cnpjCpfFornecedor;
	
	// Paginacao
	private Integer pagina;
	private Integer itens;
	
}

File: web/angular/src/app/shared/model/ParametrosConsulta.ts
export class ParametrosConsulta {

    // filtros
    public id: number;
    public nome: string;
    public idLegislatura: number[] = [];
    public siglaUf: string[] = [];
    public siglaPartido: string[] = [];
    public siglaSexo: string;
    public ano: number[] = [];
    public mes: number[] = [];
    public cnpjCpfFornecedor: string;

    // paginacao
    public itens: number = 20;
    public pagina: number = 1;
}
